hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMClientRequestUtils.java

```diff
@@ -47,4 +60,40 @@ public static void checkClientRequestPrecondition(
           OMException.ResultCodes.INTERNAL_ERROR);
     }
   }
+
+  public static boolean isSnapshotBucket(OMMetadataManager omMetadataManager,
+      OmKeyInfo keyInfo) throws IOException {
+
+    boolean status;
+    TableIterator<String, ? extends Table.KeyValue<String, SnapshotInfo>>
+        iterator = omMetadataManager.getSnapshotInfoTable().iterator();
+    String dbSnapshotBucketKey = omMetadataManager.getBucketKey(
+        keyInfo.getVolumeName(), keyInfo.getBucketName())
+        + OM_KEY_PREFIX;
+
+    iterator.seek(dbSnapshotBucketKey);
+    status = iterator.hasNext() && iterator.next().getKey()
```

Comment:
Curious if we need to check the volume and bucket in case of legacy bucket type. Similar to https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java#L1240

Reply:
We are also checking the snapshotDbKey which is `volume/bucket/snapName` along with the `seek()`, so it should be fine.